graphene_plugin.py

```diff
@@ -686,38 +697,19 @@ def process_gql_schema(ctx: AttributeContext) -> Type:
 
             assert isinstance(ctx.type, Instance)
             object_info = self._graphene_objects[ctx.type.type.fullname]
+            all_fields = object_info.all_fields if isinstance(object_info, ObjectTypeInfo) else object_info.fields
 
-            for field in object_info.fields.values():
-                resolver = object_info.resolvers.get(field.name)
+            # Check that resolver methods are annotated with the correct types
+            for resolver in object_info.resolvers.values():
+                gql_field = all_fields.get(resolver.field_name)
```

Comment:
Then I decided to invert/split the type-checking logic a bit. The first thing we're checking is that resolvers have the correct type annotations. Given that we're type-checking _resolvers_, I thought it'd make more sense to iterate over resolvers (and then look up the corresponding field definition) rather than iterate over fields.

While looping over resolvers, we lookup `Field()`s via `.all_fields` so that both the `ObjectType`'s own `Field()`s, as well as the ones defined on its `Interface`s are checked.